CMakeLists.txt

```diff
@@ -75,9 +79,10 @@ file(GLOB S3_HEADERS
 
 file(GLOB S3_SRC
         ${AWS_S3_ROOT_SRC}
-        ${AWS_S3_MODEL_SRC}
-        ${AWS_S3_EXTERNAL_SRC}
         )
+if (AWS_ENABLE_S3_ENDPOINT_RESOLVER)
+    list(APPEND S3_SRC ${AWS_S3_ENDPOINT_RESOLVER_SRC})
```

Comment:
I don't understand how this works, just throw out my question:
So, this will only compile the `source/s3_endpoint_resolver/*.c` when the cmake config is set. What about the header file? Will Cmake not compile the header files as well?

Reply:
Yes, it will only include the source when the CMake configuration is set. The header file is always included to simplify things. However, we warn users not to use it unless it's enabled.